test/dynamo/test_repros.py

```diff
@@ -872,8 +872,9 @@ def test_longformer_chunk(self):
         self.assertTrue(same(opt_fn(input1), correct1))
         self.assertTrue(same(opt_fn(input2), correct2))
 
-        self.assertEqual(cnt.frame_count, ifdyn(1, 2))
-        self.assertEqual(cnt.op_count, ifdyn(19, 4))
+        # Dyn recompiles are due to changes in hidden_state (Should we be guarding on this?)
```

Comment:
Err, I don't understand; if you guard on it, wouldn't you recompile more?

Reply:
Yes, this increased our number of recompiles from 0 to 3

Reply:
Ah, I read the question mark as "should we do this?" but it is the reverse "should we not guard on this?"

Reply:
Gotcha. The real question is - should we be guarding on internal state? I think so.

Reply:
But the maximum number of guards really should only be 2, because there are only two distinct sizes passed in this test. That there are four implies something has gone wrong.

Reply:
We aren't graph breaking on inputs we are graph breaking on internal state. `hidden_state` changes because its internal state. All the guards are on `hidden_state`

Changing this test to:

```
        self.assertTrue(same(opt_fn(input1), correct1))
        self.assertTrue(same(opt_fn(input1), correct1))
        self.assertTrue(same(opt_fn(input1), correct1))
        self.assertTrue(same(opt_fn(input1), correct1))
```

Still produces 4 frames.